src/index.js

```diff
@@ -33,56 +35,68 @@ export type ActionLoggerOptionsType = {
 
 const isEmptyObject = (obj = {}) => !Object.keys(obj).length;
 
-async function sendToEndpoint(endpoint: EndpointType, state: Object, log: Object): Promise<Response> {
-  let headers = {
+function createDefaultEndpointFunction(endpoint: HttpEndpointConfig): CustomEndpointFunction {
+  const defaultHeaders = {
     'Accept': 'application/json',
     'Content-Type': 'application/json',
   };
-
-  if (endpoint.headers) {
-    const mergedHeaders = {...endpoint.headers};
-
-    // go through each header. if function - call and replace value
-    for (const key in endpoint.headers) {
-      if (typeof endpoint.headers[key] === 'function') {
-        mergedHeaders[key] = (endpoint.headers[key].call(undefined, state): string);
-      } else if (typeof endpoint.headers[key] === 'boolean') {
-        mergedHeaders[key] = endpoint.headers[key].toString(); // booleans need to be manually converted
+  const headers = endpoint.headers;
+  const includeCredentials = endpoint.includeCredentials;
+  const transform = endpoint.transformFunction;
+  const uri = endpoint.uri;
+
+  return (log: Object, state: Object): Promise<bool> => {
+    let finalHeaders = {};
+    let finalLog = log;
+
+    if (headers) {
+      const mergedHeaders = {...headers};
+
+      // go through each header. if function - call and replace value
+      for (const key in endpoint.headers) {
+        if (typeof endpoint.headers[key] === 'function') {
+          mergedHeaders[key] = (endpoint.headers[key].call(undefined, state): string);
```

Comment:
Could do this in a future PR, but a try/catch here feel warranted. Maybe just set the header to something like `"Exception generating header"` if it fails